iothub/device/src/ClientFactory.cs

```diff
@@ -138,11 +133,6 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
                     throw new ArgumentException("certificate must be present in DeviceAuthenticationWithX509Certificate");
                 }
 
-                if (!connectionStringBuilder.Certificate.HasPrivateKey)
-                {
```

Comment:
Can we try to check why did we have this in the first place till now ? I agree private key shouldn't be presented. Also how did it work with this check until now?

Reply:
IoT hub doesn't care about the private key; all they require is the thumbprint for self-signed certificates, and the certificate chain for CA authenticated devices. If the device presents a certificate whose private key it doesn't own, then the authentication will fail when completing the TLS handshake.

Reply:
Also, here we weren't actually requiring the private key as a string/ some exportable format; we simply validated that the certificate being passed in has the private key available. This is usually true for most windows based systems; but other OS might be having different constraints.